libsql/src/sync.rs

```diff
@@ -309,7 +309,7 @@ impl SyncContext {
     async fn read_metadata(&mut self) -> Result<()> {
         let path = format!("{}-info", self.db_path);
 
-        if !std::fs::exists(&path).map_err(SyncError::io("metadata file exists"))? {
+        if !Path::new(&path).exists() {
```

Comment:
This hides potential errors. Is there an API that does not hide them and works for older build?

Reply:
[try_exists](https://doc.rust-lang.org/std/path/struct.Path.html#method.try_exists) seems to do the job and is there since 1.63